### PR TITLE
M2-724 Builder - Activity Items tab - Item configuration - Color palette

### PR DESCRIPTION
**Pull request by muhurov**

### CODE REVIEW

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.tsx [hunk 1]
```diff
@@ -58,6 +59,7 @@ export const ItemConfiguration = () => {
       name: '',
       body: '',
       settings: [],
+      palette: '',
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.tsx:62]: pls don't set this property by default, it's not obligatory field

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/Settings/ColorPalette/ColorPalette.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const handleCollapse = () => setIsExpanded((prevExpanded) => !prevExpanded);
+  const handleRemovePalette = () => {
+    setValue(
+      'settings',
+      settings?.filter(
+        (setting: ItemConfigurationSettings) =>
+          setting !== ItemConfigurationSettings.HasColorPalette,
+      ),
+    );
+    setValue('palette', '');
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/Settings/ColorPalette/ColorPalette.tsx:25]: don't set palette on remove, maybe it's better to unregister this field

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/Settings/ColorPalette/ColorPalettePicker/ColorPalettePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }, [palette]);
+
+  return (
+    <StyledColorPalettePickerContainer>
+      <RadioGroupController control={control} name="palette" options={RADIO_GROUP_OPTIONS} />
+      <StyledFlexTopCenter sx={{ gap: '1.2rem' }}>
+        {SELECTION_OPTIONS_COLOR_PALETTE.map(({ name, colors }) => (
+          <StyledFlexColumn key={`palette-${name}`} sx={{ flexGrow: 1, gap: '0.1rem' }}>
+            {colors.map((hex) => (
+              <StyledPaletteColorBox
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/Settings/ColorPalette/ColorPalettePicker/ColorPalettePicker.tsx:46]: pls use <Styled... />

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionOption/ColorPicker/ColorPicker.tsx [hunk 4]
```diff
@@ -10,29 +10,39 @@ export const ColorPicker = <T extends FieldValues>({
...
-          }}
-        >
-          <ChromePicker disableAlpha={true} color={value} onChangeComplete={onChange} />
-        </Popover>
-      )}
+      render={({ field: { onChange, value } }) => {
+        const handleChange: ColorChangeHandler = (...args) => {
+          onChange(...args);
+
+          setValue('palette', '');
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionOption/ColorPicker/ColorPicker.tsx:25]: Should not we use 'palette' as a paletteName property for the component?

Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionOption/ColorPicker/ColorPicker.tsx:25]: maybe also use unregister here?

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.hooks.ts [hunk 5]
```diff
@@ -68,6 +69,10 @@ export const useSettingsSetup = ({ control, setValue, getValues, watch }: Settin
     !hasAlerts && removeAlert();
   }, [hasAlerts]);
 
+  useEffect(() => {
+    setValue('palette', '');
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.hooks.ts:73]: maybe also use unregister here?

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionOption/ColorPicker/ColorPicker.tsx (hunk 4 above)
Comment (muhurov) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionOption/ColorPicker/ColorPicker.tsx]: done

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**